Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Lib/DriverScore.hs

```diff
@@ -90,8 +92,9 @@ eventPayloadHandler merchantOpCityId DST.OnDriverCancellation {..} = do
       | otherwise = a
 eventPayloadHandler merchantOpCityId DST.OnRideCompletion {..} = do
   mbDriverStats <- B.runInReplica $ DSQ.findById (cast driverId) -- always be just because stats will be created at OnNewRideAssigned
+  fareParameter <- maybe (pure Nothing) (B.runInReplica . FPQ.findById) ride.fareParametersId
```

Comment:
bro, fareParameters are already fetch from where this function is called. You don't need to fetch it again.